### PR TITLE
Replace PropTypes with TypeScript types

### LINKED ISSUE #403: replace propTypes with typescript implementation
```ts
import { Tag, TagProps } from '*****';

const COLOR = 'blue';

interface SampledTagProps extends TagProps {
  show?: boolean;
}

export function SampledTag({ show, ...props }: SampledTagProps) {
  return (show !== undefined ? show : false) ? (
    <Tag color={COLOR} className="datadiff-sampled-tag" {...props}>
      sampled
    </Tag>
  ) : null;
}
```

### PR DESCRIPTION
**Pull request by kofta999**
# The Pull Request is ready

- [x] fixes #403 
- [ ] all actions are passing
- [x] only fixes a single issue

## Overview

Replaces runtime type checking introduced with PropTypes package with static type checking using TypeScript

Also added keys to the list items in breadcrumbs component to fix React warning about not using keys in list items
See: https://sentry.io/answers/unique-key-prop/ 

## Website

- [x] mobile view is usable
- [x] desktop view is usable
- [x] no a-tags are used directly (NavLink, MailLink, ExternalLink instead)
- [ ] all new texts are added to the translation files (at least the english one)
- [ ] tests have been added (if required)
- [ ] shared code has been extracted in a different file

## Notes

Please check if using `name` as a key is unique enough, couldn't think of anything else.
Didn't intend to add the keys in this PR but I found it by chance while checking if everything is fine in the website

### CODE REVIEW

**Comment by kofta999** [comment]
I'm not sure why did code coverage decrease even I haven't changed anything in most components' code, so will mark for review and wait for the maintainer's opinion on this.

**Comment by Idrinth** [comment]
> I'm not sure why did code coverage decrease even I haven't changed anything in most components' code, so will mark for review and wait for the maintainer's opinion on this.

that happens, because you defined the interfaces in the file they are used in to my knowledge.

**Comment by kofta999** [comment]
So should I create a separate interfaces.ts file and add all of them there? I doubt that's what normally happens with component props though

Or it's fine for coverage to decrease? It's up to you anyway.

File: website/src/components/breadcrumbs.tsx [hunk 1]
```diff
@@ -17,7 +20,7 @@ const build = (
 ) => {
   const name = parts[index] === '' ? 'home' : parts[index];
   if (index === parts.length - ONE) {
-    nav.push(<li>
+    nav.push(<li key={name}>
```
Comment (Idrinth) [inline website/src/components/breadcrumbs.tsx:23]: why does the li get a key property?

**Review by Idrinth — CHANGES_REQUESTED** [review]

**Comment by Idrinth** [comment]
> So should I create a separate interfaces.ts file and add all of them there? I doubt that's what normally happens with component props though
> 
> Or it's fine for coverage to decrease? It's up to you anyway.

sorry about the late reply, the drop is fine - we'll recover from that

File: website/src/components/breadcrumbs.tsx (hunk 1 above)
Comment (kofta999) [reply website/src/components/breadcrumbs.tsx:23]: I did it to deal with this React warning as described above

image

If the warning is not that important I'll gladly remove it

**Pull request merged**